### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by alvin-garcia** [pencil-code/pencil-code#28]
Hi, I am trying to run a test case of your code for my MHD graduate course. I ultimately want to run the test case in the tokamak folder, so I would appreciate any pointers on that as well please. I am having a compilation issue that is taking me too much time to solve. Can you please help?
[pencil_code_output.pdf](https://github.com/pencil-code/pencil-code/files/2650978/pencil_code_output.pdf)

**Comment by mrheinhardt** [comment]
Hi Alvin,


thank you for your interest in the Pencil Code. Your problem in the build is apparently due to the lack of the system calls

dlsym, dlerror, dlclose


on your system. As an ad-hoc measure, please edit syscalls_ansi.c as follows:

 return dlsym(handle, symbol); -> return NULL;

char *error=dlerror(); -> char *error=NULL;

dlclose(handle); -> //dlclose(handle);

Update the code and run. However, the stored reference data (reference.out) will not be
reproduced. The sample has not been updated for a while, so we have to have a closer look
into this.

Best regards,
Matthias

**Comment by alvin-garcia** [comment]
Hi Matthias,

Thank you for responding. I have made those changes, but I still get the same compile error. I should mention that I am following the direction of the quick_start.pdf from your website.

[penicl_code_output2.pdf](https://github.com/pencil-code/pencil-code/files/2654319/penicl_code_output2.pdf)

**Comment by mrheinhardt** [comment]
Hi again,


I overlooked one instance of dlerror:


printf("error = %s\n", dlerror());


needs also to be set on comment. But dlsym and dlclose should not have appeared in your error messages, so I don't understand it.

After compilation of syscalls_ansi.c the object syscalls_ansi.o should not contain dlsym dlerror or dlclose as unresolved names,

so, e.g.,


nm syscalls_ansi.o | grep dlclose


should give


0000000000000230 T dlclose_c_


and nothing more.


Best,

Matthias

________________________________
Von: Alvin Garcia <notifications@github.com>
Gesendet: Donnerstag, 6. Dezember 2018 19:23:05
An: pencil-code/pencil-code
Cc: mrheinhardt; Comment
Betreff: Re: [pencil-code/pencil-code] Compilation error (#28)


Hi Matthias,

Thank you for responding. I have made those changes, but I still get the same compile error. I should mention that I am following the direction of the quick_start.pdf from your website.

penicl_code_output2.pdf<https://github.com/pencil-code/pencil-code/files/2654319/penicl_code_output2.pdf>

**Comment by alvin-garcia** [comment]
Hey Matthias,

Great, I have gotten an output and I've followed the quick start guide again for the tokamak sample case.

It seems like I've got data in my folder, but I am unsure on how to visualize it. I've followed your python directions, but I am unsure on how to make sense of the variable definitions.

I will be talking about the Grad-Shafranov equation in my final project. I would like to have some information on the parameters of the pencil code, i.e. the assumptions made, plasma profiles, EM fields, etc. Perhaps, showing the magnetic axis, contour lines, poloidal and toroidal flux functions etc. would also be good.

What variables should I be plotting to illustrate some of these key concepts and how can I read any of these from `var`? I am familiar with Matplotlib, so I just need to figure out how get data in the form of arrays.

[penicl_code_output3.pdf](https://github.com/pencil-code/pencil-code/files/2654697/penicl_code_output3.pdf)

**Comment by mrheinhardt** [comment]
Hi,


the setup in a nutshell:


Quantities: velocity (uu), log density (lnrho), magnetic vector potential A (aa)

Domain: hollow cylinder, 1<=r<=3, -1<=z<=1, full phi

BCs: periodic in z,phi,

        at r-boundaries: impenetrable  for flow, dA_r/dr=0,

                                               A_phi, A_z frozen (initial values enforced)

Initial condition: see src/initial_condition/fluxring_cylindrical.f90

Driving: emf in phi direction propto (r-r_inner)(r_outer-r)(z-z_bottom)(z_z_top)

Fluid: homogeneous; constant nu, eta, see run.in

Equations: isothermal compressible MHD, see manual


I'm not using Python. With IDL, the snapshot var.dat would be read by


pc_read_var, object=<name>


<name> (choosable) is a structure containing the uu, lnrho, aa fields on the grid, surrounded by a 3 points thick ghost layer (can be removed by option /trimall to the above call). To get also the magnetic field, use /bb in the call, /jj for current density.


help, v


would give you all information about the structure of the data contained.


Note that in the sample the resolution might not be appropriate.


Cheers,

Matthias



________________________________
Von: Alvin Garcia <notifications@github.com>
Gesendet: Donnerstag, 6. Dezember 2018 21:20:32
An: pencil-code/pencil-code
Cc: mrheinhardt; Comment
Betreff: Re: [pencil-code/pencil-code] Compilation error (#28)


Hey Matthias,

Great, I have gotten an output and I've followed the quick start guide again for the tokamak sample case.

It seems like I've got data in my folder, but I am unsure on how to visualize it. I've followed your python directions, but I am unsure on how to make sense of the variable definitions.

I will be talking about the Grad-Shafranov equation in my final project. I would like to have some information on the parameters of the pencil code, i.e. the assumptions made, plasma profiles, EM fields, etc. Perhaps, showing the magnetic axis, contour lines, poloidal and toroidal flux functions etc. would also be good.

What variables should I be plotting to illustrate some of these key concepts and how can I read any of these from var? I am familiar with Matplotlib, so I just need to figure out how get data in the form of arrays.

penicl_code_output3.pdf<https://github.com/pencil-code/pencil-code/files/2654697/penicl_code_output3.pdf>
